src/main/java/net/dries007/holoInventory/client/GroupRenderer.java

```diff
@@ -148,7 +150,25 @@ public void renderItems(List<ItemStack> itemStacks) {
      */
     private void renderItem(ItemStack itemStack, int column, int row) {
         if (itemStack == null) return;
-        ItemStack renderStack = itemStack.copy();
+        ItemStack renderStack = null;
+
+        // TF maps uses a different item entity renderer which causes issues with holo and Angelica
+        // couldn't figure a way to get it to work in TF but not break anything so lets go with this
+        // solution: render empty map type instead
+        if (HoloInventory.isTFLoaded) {
+            // max stack size is 1 for the original map so lets hardcode it
+
+            if (itemStack.getUnlocalizedName().equalsIgnoreCase("item.magicmap")) {
+                renderStack = new ItemStack(TFItems.emptyMagicMap, 1, 0);
+            } else if (itemStack.getUnlocalizedName().equalsIgnoreCase("item.mazemap")) {
+                renderStack = new ItemStack(TFItems.emptyMazeMap, 1, 0);
+            } else if (itemStack.getUnlocalizedName().equalsIgnoreCase("item.oremap")) {
```

Comment:
It looks for the class when it executes it i believe

Reply:
> Comparing the item of the item stack would require calling `getItem()` which for GT items is a lot of extra code to run in a hot path.

Can you explain why? This is the first time I hear this. If you don't want to call `getItem()`, you could cache an ItemStack for each of the maps and use `ItemStack#isItemEqual(ItemStack)` instead.

Reply:
I think it was the same reason why we use the feather for getting and setting an item's metadata. It was just faster, but yeah, i can change it to use isItemEqual instead.

Reply:
I was surprised to hear that because `ItemStack#getItem()` only does a null-check and call `RegistryDelegate#get()` and `ItemStack` is a final class so GT can't just extend and override. We generally use the feather item because it's just a "dumb" item and doesn't override `Item#setDamage(ItemStack, int)`. We could just as well use e.g. the flint item.